File: except/except/file_read.c
#include <stdbool.h>
#include <stdio.h>

bool process_file(char *name)
{
        FILE *f;
        f = fopen(name, "rb");
        if (f == NULL) {
                return false;
        }
        fclose(f);
        return true;
}
    
int
main(int argc, char *argv[])
{
        (void) argc;
        char *fname = argv[1];
        bool success;
        success = process_file(fname);
        if (!success) {
                printf("Could not read file %s\n", fname);
        }
}

File: except/except/file_read.py
#!/bin/env python

import sys



fname = sys.argv[1]

#
#     process_file
#
def process_file(name):
    f = open(fname, 'rb')
    for l in f.readlines():
        print l
    
#
#     main
#
try:
    process_file(fname)
except IOError:
    print "Could not read file:", fname
    sys.exit()

